#! /bin/sh /usr/share/dpatch/dpatch-run
## 09_maxfontsize.dpatch by Roland Rosenfeld <roland@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: For fonts >=42pt use the given font size in TeX instead of
## DP: restricting this to 42pt. This allows to create posters using
## DP: \usepackage{type1cm}. See
## DP: http://linuxgazette.vlsm.org/issue96/artime.html and Bug#343139

@DPATCH@
diff -urNad transfig~/LATEX.AND.XFIG transfig/LATEX.AND.XFIG
--- transfig~/LATEX.AND.XFIG
+++ transfig/LATEX.AND.XFIG
@@ -55,7 +55,7 @@
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% LaTeX Preamble %%%%%%%%%%%%%%%%%%%%%%%%%
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
  
-\documentstyle[12pt,bezier,amstex]{article}  % include bezier curves
+\documentclass[12pt,bezier,amstex]{article}  % include bezier curves
 \renewcommand\baselinestretch{1.0}           % single space
 \pagestyle{empty}                            % no headers and page numbers
 \oddsidemargin -10 true pt      % Left margin on odd-numbered pages.
@@ -72,7 +72,9 @@
 \usepackage{epsfig}
 
 \usepackage{graphicx}           % Graphics package
- 
+
+\usepackage{type1cm}            % Necessary for fonts bigger than 42pt
+
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% Document Beginning %%%%%%%%%%%%%%%%%%%%%
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
diff -urNad transfig~/fig2dev/dev/texfonts.h transfig/fig2dev/dev/texfonts.h
--- transfig~/fig2dev/dev/texfonts.h
+++ transfig/fig2dev/dev/texfonts.h
@@ -86,6 +86,7 @@
 
 #define TEXFONTSIZE(S)	(texfontsizes[((S) <= MAXFONTSIZE) ? (int)(round(S))\
 				      				: (MAXFONTSIZE-1)])
+#define TEXFONTSIZE(S)	(((S) <= MAXFONTSIZE) ? texfontsizes[(int)(round(S))] : (S))
 #define TEXFONTMAG(T)	TEXFONTSIZE(T->size*(rigid_text(T) ? 1.0 : fontmag))
 
 void setfigfont( F_text *text );		/* genepic.c */
